go-controller/pkg/node/healthcheck.go

```diff
@@ -93,6 +95,29 @@ func countLocalEndpoints(ep *kapi.Endpoints, nodeName string) int {
 	return num
 }
 
+func countHostNeworkedEndpoints(ep *kapi.Endpoints, nodeName string) int {
```

Comment:
I think you might be able to implement this more easily - HostNetwork Pods always have the "primary" IP of the Node as their IP. So you can probably just try and search for endpoints as "IP in NodeIPs"

Reply:
Updated this function to return a bool and use the new nodeIP manager